llvm/lib/Support/APInt.cpp

```diff
@@ -93,7 +88,7 @@ void APInt::initFromArray(ArrayRef<uint64_t> bigVal) {
     U.VAL = bigVal[0];
   else {
     // Get memory, cleared to 0
-    U.pVal = getClearedMemory(getNumWords());
+    U.pVal = new WordType[getNumWords()]();
```

Comment:
Similar here. If bigVal.size() is smaller than getNumWords() you will not initialize some words.

Reply:
The default-initialization of `WordType` in this case should be zero-initialization.